test/IRGen/objc_direct.swift

```diff
@@ -0,0 +1,40 @@
+// RUN: %target-swift-frontend -import-objc-header %S/Inputs/objc_direct.h -emit-ir -o - %s | %FileCheck --check-prefix=CHECK-IR %s
+// RUN: %empty-directory(%t)
+// RUN: %clang -isysroot %sdk %S/Inputs/objc_direct.m -c -o %t/objc_direct.o
+// RUN: %target-build-swift -import-objc-header %S/Inputs/objc_direct.h -Xlinker %t/objc_direct.o %s -o %t/a.out
+// RUN: %target-run %t/a.out | %FileCheck %s
```

Comment:
I think it would be more consistent to split this up into an IRGen test and an execution test in test/Interpreter. The execution test might need to be guarded by an availability check if objc_direct methods don't backward deploy (I don't know if they do or not).